src/lib/markbind/src/patches/htmlparser2.js

```diff
@@ -129,6 +148,189 @@ Tokenizer.prototype._stateText = function(c){
 	}
 };
 
+
+Tokenizer.prototype.specialTagNames = [
+	'script',
+	'style',
+];
+
+/**
+ * Checks whether the token matches one of the first characters of the special tags,
+ * and initialises the _matchingSpecialTags array with the matches' indexes if so.
+ */
+Tokenizer.prototype.isFirstCharacterSpecialTagCharacter = function(c) {
+	this._matchingSpecialTags = [];
+	const numSpecialTags = this.specialTagNames.length;
+	const lowerCaseChar = c.toLowerCase();
+	for (let j = 0; j < numSpecialTags; j += 1) {
+		if (lowerCaseChar === this.specialTagNames[j][0]) {
+			this._matchingSpecialTags.push(j)
+		}
+	}
+
+	if (this._matchingSpecialTags.length > 0) {
+		this._nextSpecialTagMatchIndex = 1;
+		return true
+	}
+	return false;
+};
+
+/**
+ * Processes the _matchingSpecialTags array, filtering out previous match indexes
+ * that do not match the current token.
+ * If one of the previous matches successfully matched, the match index is returned.
+ */
+Tokenizer.prototype.processSpecialFunctions = function(c) {
+	const matchingSpecialTags = [];
+	const numMatchingTags = this._matchingSpecialTags.length;
```

Comment:
Minor: Could `_matchingSpecialTags` be renamed to `_matchingSpecialTagsIndex`? It makes the operation below a little less confusing.

Reply:
Updated